src/libraries/System.Private.CoreLib/src/System/Buffer.cs

```diff
@@ -194,9 +194,6 @@ private static void _BulkMoveWithWriteBarrier(ref byte destination, ref byte sou
         {
             Debug.Assert(byteCount > BulkMoveWithWriteBarrierChunk);
 
-            if (Unsafe.AreSame(ref source, ref destination))
-                return;
-
             // This is equivalent to: (destination - source) >= byteCount || (destination - source) < 0
```

Comment:
```suggestion
            if (Unsafe.AreSame(ref source, ref destination))
                return;

            // This is equivalent to: (destination - source) >= byteCount || (destination - source) < 0
```
I would keep this check. It will save us running the loop below for identical buffers.